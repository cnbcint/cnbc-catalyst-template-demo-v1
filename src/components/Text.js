import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Pdrop = styled.p`
  color: ${(props) => props.color};
  font-size: ${(props) => (props.highlight ? "1.4em" : "1.2rem")};
  line-height: ${(props) => (props.highlight ? "140%" : "125%")};
  text-align: ${(props) => props.textAlign};
  max-width: 600px;
  margin:0 auto;
  &::first-letter {
    color: ${(props) => props.dropCapColor};
    padding: 0 0;
    margin: 0.5rem 1rem 0 0;
    font-size: 5.5rem;
    font-weight: bold;
    float: left;
    line-height: 4rem;
  }
`;
const P = styled.p`
  color: ${(props) => props.color};
  font-size: ${(props) => (props.highlight ? "1.4rem" : "1.2rem")};
  line-height: ${(props) => (props.highlight ? "140%" : "125%")};
  text-align: ${(props) => props.textAlign};
  max-width: 600px;
  margin:0 auto;

`;
/** 
 Text Block will add a paragraph element to it's children. Do not add P inside this element to avoid double nested P element
*/
export default function Text(props) {
  if (props.dropCap) {
    return (
      <Pdrop
        highlight={props.highlight}
        dropCapColor={props.dropCapColor ? props.dropCapColor : props.color}
        color={props.color}
        textAlign={props.textAlign}
        style={props.style}
        className={props.className}
      >
        {props.children}
      </Pdrop>
    );
  }
  return (
    <P
      highlight={props.highlight}
      color={props.color}
      style={props.style}
      textAlign={props.textAlign}
      className={props.className}
    >
      {props.children}
    </P>
  );
}

Text.defaulProps = {
  dropCap: false,
  color: "inherit",
  dropCapColor: "",
  highlight: false,
  className: "",
  textAlign: "left",
};

Text.propTypes = {
  /**
   Will Increase font size to highlight the text
   */
  highlight: PropTypes.bool,
  /**
   Will Increase font size of the first letter of the paragraph to achive DropCap 
   */
  dropCap: PropTypes.bool,
  /**
   CSS override for font color
   */
  color: PropTypes.string,
  /**
   CSS text aligment
   */
  textAlign: PropTypes.oneOf(["left", "right", "center"]),
  /**
   CSS color for drop cap
   */
  dropCapColor: PropTypes.string,
  /**
  React CSS format to override rules has required by design
  */
  style: PropTypes.object,
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
};
