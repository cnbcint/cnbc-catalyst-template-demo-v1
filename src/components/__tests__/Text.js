import React from "react";
import Text from "../Text";
import { render } from "@testing-library/react";
import "jest-styled-components";
import { css } from "styled-components";

describe("Text", () => {
  it("Renders correctly", () => {
    const { container } = render(<Text>Testing Content</Text>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders correctly dropCap", () => {
    const { container } = render(<Text dropCap>Testing Content</Text>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders correctly highlighted", () => {
    const { container } = render(<Text highlight>Testing Content</Text>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders the correct content", () => {
    const { container } = render(<Text>Testing Content</Text>);
    expect(container.firstChild.textContent).toMatch("Testing Content");
  });

  it("Passes the class name", () => {
    const { container } = render(
      <Text className="testClass">Testing Title</Text>
    );
    expect(container.firstChild.classList.contains("testClass")).toBe(true);
  });

  it("Set The right Color", () => {
    const { container } = render(<Text color="blue">Testing Title</Text>);
    expect(container.firstChild).toHaveStyleRule("color", "blue");
  });

  it("Set The right dropcap color", () => {
    const { container } = render(
      <Text dropCap dropCapColor="blue">
        Testing Title
      </Text>
    );
    expect(container.firstChild).toHaveStyleRule("color", "blue", {
      modifier: css`&::first-letter`,
    });
  });
});
