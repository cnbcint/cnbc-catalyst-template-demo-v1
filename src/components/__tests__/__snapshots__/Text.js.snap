// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Renders correctly 1`] = `
.c0 {
  font-size: 1.2rem;
  line-height: 125%;
}

<p
  class="c0"
>
  Testing Content
</p>
`;

exports[`Text Renders correctly dropCap 1`] = `
.c0 {
  font-size: 1.2rem;
  line-height: 125%;
}

.c0::first-letter {
  padding: 0 0;
  margin: 0.5rem 1rem 0 0;
  font-size: 5.5rem;
  font-weight: bold;
  float: left;
  line-height: 4rem;
}

<p
  class="c0"
>
  Testing Content
</p>
`;

exports[`Text Renders correctly highlighted 1`] = `
.c0 {
  font-size: 1.4rem;
  line-height: 140%;
}

<p
  class="c0"
>
  Testing Content
</p>
`;
